p2p/module.go

```diff
@@ -31,6 +32,29 @@ type p2pModule struct {
 	network typesP2P.Network
 }
 
+// HandleEvent implements modules.P2PModule
+func (m *p2pModule) HandleEvent(message *anypb.Any) error {
+	switch message.MessageName() {
+	case messaging.BeforeHeightChangedEventType:
+		msg, err := codec.GetCodec().FromAny(message)
+		if err != nil {
+			return err
+		}
+		_, ok := msg.(*messaging.BeforeHeightChangedEvent)
+		if !ok {
+			return fmt.Errorf("failed to cast message to BeforeHeightChangedEvent")
+		}
+
+		// DISCUSS (https://github.com/pokt-network/pocket/pull/374#issuecomment-1341350786): decide if we want a pull or push model for integrating with persistence
```

Comment:
Consolidating the discussions here:
- https://github.com/pokt-network/pocket/pull/374#issuecomment-1341350786
- https://github.com/pokt-network/pocket/pull/374#issuecomment-1347532500

The main reason I was suggesting those alternate events and functions is because `getAddrBookPerHeight` solves the problem of integrating with persistence fulfilling what this ticket intended to do.

I don't currently see how we're using these two events (in this PR) so am thinking we should just remove altogether, but wanted to leverage the work already done.

Reply:
It makes total sense. However my code was there to convey a train of thought and to clarify my doubts, it was a poor-man's design spec in a way.

Now that things are clearer the P2P event handling can be removed, what I'd like to keep, if you agree, is just the events `BeforeNewHeightEvent` and `HeightChangedEvent` since they will likely be used as "hooks" later on. For now they are simply NOOP.

Reply:
To quote you previous message: 

> Where would this function be called currently? I would say that if it's not relevant to the scope of this issue we shouldn't do it here. Sounds like scope creep a little bit. No?

I'm personally not a huge fan because these events were out of scope and are not being used, though I can see where your head is at.

If I were to introduce a new event, I'd probably create a `NewViewEvent(height, round)` and/or `NewHeight(height)` because I can only speculate how we'd use `BeforeNewHeightEvent` right now and the cost to add it is low.

I don't want this to continue being a blocker, but not a fan. Let's do this:
1. If you feel strongly that `BeforeNewHeightEvent` and `HeightChangedEvent` events will be useful, let's keep it as is.
2. If you're not sure, let's do one of the following:
2.1. Remove them altogether
2.2 Just add a `NewViewEvent` (per my suggestions)